src/server.cpp

```diff
@@ -966,25 +971,27 @@ static CTimingMeas JitterMeas ( 1000, "test2.dat" ); JitterMeas.Measure(); // TE
     }
 }
 
-void CServer::DecodeReceiveDataBlocks ( const int iStartChanCnt,
+void CServer::DecodeReceiveDataBlocks ( CServer*  sServer,
+                                        const int iStartChanCnt,
                                         const int iStopChanCnt,
                                         const int iNumClients )
 {
     // loop over all channels in the current block, needed for multithreading support
     for ( int iChanCnt = iStartChanCnt; iChanCnt <= iStopChanCnt; iChanCnt++ )
     {
-        DecodeReceiveData ( iChanCnt, iNumClients );
+        sServer->DecodeReceiveData ( iChanCnt, iNumClients );
```

Comment:
I think it's because `CServer::DecodeReceiveDataBlocks` and `CServer::MixEncodeTransmitDataBlocks` can only be passed to `tpThreadPool->enqueue()` as static functions, not instance methods, and therefore do not automatically inherit a `this` pointer when called. So it has to be passed manually as the first argument, and used explicitly when calling the class methods `DecodeReceiveData` and `MixEncodeTransmitData`. I don't think it's a problem, but it possibly deserves an explanatory comment in the code.